Tools-Override/tests.targets

```diff
@@ -62,8 +65,60 @@
 
   <!-- The Code Coverage targets will override TestHost and TestCommandLine if coverage is enabled -->
   <Import Project="$(MSBuildThisFileDirectory)CodeCoverage.targets" />
+  
+  <!-- This will generate a version of the dotnet cli that runs with the live
+       build of corefx using the other components gathered from the stable 
+       version consumed from the Tools directory
+       To debug any issues set COREHOST_TRACE=1 and rerun dotnet
+  -->
+  <UsingTask TaskName="GenerateDepsJson" AssemblyFile="$(BuildToolsTaskDir)CoreFx.Tools.dll"/>  
+  <Target Name="ConstructTestingHost" Condition="!Exists('$(TestSharedFxDir)')" >
+    <ItemGroup>
+      <HostFxrFile Include="$(RuntimePath)hostfxr.dll" />
+      <HostFxrFile Include="$(RuntimePath)libhostfxr.so" />
```

Comment:
They aren't restored, I just check if any of the above existed. I can condition them if required.